source/SharedCrtResourceManager.cpp

```diff
@@ -88,11 +89,23 @@ void SharedCrtResourceManager::initializeAllocator()
     allocator = aws_mem_tracer_new(aws_default_allocator(), nullptr, AWS_MEMTRACE_BYTES, 0);
 }
 
-int SharedCrtResourceManager::buildClient()
+int SharedCrtResourceManager::buildClient(const PlainConfig &config)
 {
     // We MUST declare an instance of the ApiHandle to perform global initialization
     // of the SDK libraries
     apiHandle = unique_ptr<ApiHandle>(new ApiHandle());
+    if (config.logConfig.sdkLoggingEnabled)
+    {
+        apiHandle->InitializeLogging(config.logConfig.sdkLogLevel, config.logConfig.sdkLogFile.c_str());
```

Comment:
I think we should add permissions on this file. Or should ask user to add permissions and we should validate those permissions on DC side

Reply:
After speaking with our AppSec team and the SDK team, there aren't currently any concerns around securing/validating the permissions of the SDK log file